### DISCUSSION TITLE
Can I disable whatever inhibits the host sleeping/turning displays off while a client is connected?

### DISCUSSION THREAD

**Discussion by alydevs** [LizardByte/community#447] [Sunshine]
Ideally I'd like the normal Windows inactivity timeout to happen, turning the screen off and disconnecting the client, which can then also sleep when Moonlight is no longer keeping it awake. I'm not sure how the inactivity timeout works or how (or even if) Sunshine inhibits it, but I'm happy to PR a new checkbox in the settings if it's straightforward.

**Comment by cgutman** [comment]
Yeah, we do inhibit it on Windows because capture will fail completely if the display is turned off.

The relevant code is: https://github.com/LizardByte/Sunshine/blob/26ceec5f3c97b0130b82cb28ba53c3b4dafc8419/src/platform/windows/display_base.cpp#L148-L155

I don't mind adding an option to disable this, but we should warn users in the UI description of the checkbox that it will cause streaming to freeze or disconnect if the remote PC display goes to sleep. If you want to plumb that option up here, what you should do is remove the `ES_CONTINUOUS` flag from the invocation of `SetThreadExecutionState()` when the option is checked. That should wake the display on connection but not keep it awake past the idle timer after that.
